spec/models/user_spec.rb

```diff
@@ -0,0 +1,13 @@
+require "rails_helper"
+require "spec_helper"
+
+RSpec.describe User, type: :model do
+  describe "validations of presence and uniqueness" do
+    it { is_expected.to validate_presence_of(:username) }
+    it { is_expected.to validate_presence_of(:email) }
+    it { is_expected.to validate_presence_of(:password) }
+    subject { FactoryBot.create(:user) }
```

Comment:
Is there any impact if we move subject to top or bottom?

Reply:
no there is no specific impact even if we dont provide subject, FactoryBot creates it on its own and tests against them

Reply:
but if we provide subject then other "it" blocks under the subject line runs tests on that specific subject

Reply:
You gave the subject on line 9. So, tests from line 6/7/8 won't use that subject?